In [1]:
# imports
import requests
import os
import pandas as pd

# Foursquare

Send a request to Foursquare with a small radius (1000m) for all the bike stations in your city of choice. 

In [22]:
api_key = os.environ["FOURSQUARE_API_KEY"]


url = "https://api.foursquare.com/v3/places/search"

headers = {"accept": "application/JSON"}
headers['Authorization'] = api_key

# params = {
#     'll' : '43.3186,11.3306',
#     'radius': '1000'
# }


params_lst = [{
    'll' : '43.321600,11.327948',
    'radius': '1000'
},
{
    'll' : '43.314554,11.331021',
    'radius': '1000'
},
{
    'll' : '43.316539,11.337212',
    'radius': '1000'
},
{
    'll' : '43.320523,11.323225',
    'radius': '1000'
},
{
    'll' : '43.328983,11.321913',
    'radius': '1000'
},
{
    'll' : '43.317187,11.354256',
    'radius': '1000'
},
{
    'll' : '43.341757,11.305010',
    'radius': '1000'
},
{
    'll' : '43.343243,11.326733',
    'radius': '1000'
},
{
    'll' : '43.336999,11.300975',
    'radius': '1000'
},
{
    'll' : '43.323781,11.338891',
    'radius': '1000'
},
{
    'll' : '43.347800,11.326046',
    'radius': '1000'
},
{
    'll' : '43.332288,11.324176',
    'radius': '1000'
},
{
    'll' : '43.341445,11.312348',
    'radius': '1000'
},
{
    'll' : '43.326764,11.333702',
    'radius': '1000'
},
{
    'll' : '43.309896,11.315263',
    'radius': '1000'
},
{
    'll' : '43.317315,11.332998',
    'radius': '1000'
},
{
    'll' : '43.296856,11.334854',
    'radius': '1000'
},
{
    'll' : '43.343407,11.322849',
    'radius': '1000'
},
{
    'll' : '43.334640,11.304803',
    'radius': '1000'
}]


# response = requests.get(url, params=params, headers=headers)
# res_json = response.json()
# res_json['results']
res_json_lst = []
for params in params_lst:
    response = requests.get(url, params=params, headers=headers)
    res_json = response.json()
    res_json_lst.append(res_json)

res_json_lst[0]

{'results': [{'fsq_id': '5958042f35d3fc2a960473c7',
   'categories': [{'id': 13064,
     'name': 'Pizzeria',
     'icon': {'prefix': 'https://ss3.4sqi.net/img/categories_v2/food/pizza_',
      'suffix': '.png'}}],
   'chains': [],
   'distance': 23,
   'geocodes': {'main': {'latitude': 43.321406, 'longitude': 11.328319},
    'roof': {'latitude': 43.321406, 'longitude': 11.328319}},
   'link': '/v3/places/5958042f35d3fc2a960473c7',
   'location': {'address': 'Viale Curtatone 18',
    'admin_region': 'Toscana',
    'country': 'IT',
    'cross_street': 'Via Tozzi',
    'formatted_address': 'Viale Curtatone 18 (Via Tozzi), 53100 Siena Toscana',
    'locality': 'Siena',
    'postcode': '53100',
    'region': 'Toscana'},
   'name': 'Pizzeria Alle Scalette',
   'related_places': {},
   'timezone': 'Europe/Rome'},
  {'fsq_id': '4cb0b499562d224b78d31888',
   'categories': [{'id': 16041,
     'name': 'Plaza',
     'icon': {'prefix': 'https://ss3.4sqi.net/img/categories_v2/parks_outdoors/plaza_',

Parse through the response to get the POI (such as restaurants, bars, etc) details you want (ratings, name, location, etc)

In [23]:
res_lst = []
for res_json in res_json_lst:
    result = res_json['results']
    for i in range(len(result)):
        dict = {}
        dict['name'] = result[i]['name']
        dict['address'] = result[i]['location']['address']
        if result[i]['categories']:
            dict['category'] = result[i]['categories'][0]['name']
        res_lst.append(dict)

res_lst

[{'name': 'Pizzeria Alle Scalette',
  'address': 'Viale Curtatone 18',
  'category': 'Pizzeria'},
 {'name': 'Piazza Giacomo Matteotti',
  'address': 'Piazza Giacomo Matteotti',
  'category': 'Plaza'},
 {'name': 'Basilica di San Domenico',
  'address': 'Piazza San Domenico',
  'category': 'Church'},
 {'name': 'Caffè La Piazzetta',
  'address': 'Via Montanini 52',
  'category': 'Bar'},
 {'name': 'Consorzio Agrario di Siena',
  'address': 'Via Pianigiani Giuseppe 9',
  'category': 'Grocery Store'},
 {'name': 'Il Masgalano',
  'address': 'Via Camporegio 1',
  'category': 'Pizzeria'},
 {'name': 'Il Pomodorino',
  'address': 'Via Camporegio 13',
  'category': 'Pizzeria'},
 {'name': 'Zest Ristorante & Winebar',
  'address': "Costa di Sant'Antonio 13",
  'category': 'Wine Bar'},
 {'name': 'Antico Bar La Lizza',
  'address': 'Piazza Antonio Gramsci 10',
  'category': 'Bar'},
 {'name': 'Ristorante San Domenico',
  'address': 'Via Camporegio 17',
  'category': 'Restaurant'},
 {'name': 'La Taverna

Put your parsed results into a DataFrame

In [24]:
df_foursquare = pd.DataFrame(res_lst)
df_foursquare

name                    address  \
0               Pizzeria Alle Scalette         Viale Curtatone 18   
1             Piazza Giacomo Matteotti   Piazza Giacomo Matteotti   
2             Basilica di San Domenico        Piazza San Domenico   
3                   Caffè La Piazzetta           Via Montanini 52   
4           Consorzio Agrario di Siena  Via Pianigiani Giuseppe 9   
..                                 ...                        ...   
185  Parrocchia Beato Bernardo Tolomei     Via Tolomei Bernardo 1   
186                 Dr. Vigni Farmacia       Via Quinto Settano 1   
187                     Poste Italiane      Via Savina Petrilli 2   
188                         Bar Giglio           Via Colombini 32   
189                  Bovicelli Roberto           Via Gallerani 23   

          category  
0         Pizzeria  
1            Plaza  
2           Church  
3              Bar  
4    Grocery Store  
..             ...  
185         Church  
186      Drugstore  
187    Post Office  
188            Bar  
189        Dentist  

[190 rows x 3 columns]

# Yelp

Send a request to Yelp with a small radius (1000m) for all the bike stations in your city of choice. 

In [17]:
yelp_params_lst = [{
    'latitude' : '43.321600',
    'longitude' : '11.327948',
    'radius': '1000'
},
{
    'latitude' : '43.314554',
    'longitude': '11.331021',
    'radius': '1000'
},
{
    'latitude' : '43.316539',
    'longitude': '11.337212',
    'radius': '1000'
},
{
    'latitude' : '43.320523',
    'longitude': '11.323225',
    'radius': '1000'
},
{
    'latitude' : '43.328983',
    'longitude': '11.321913',
    'radius': '1000'
},
{
    'latitude' : '43.317187',
    'longitude': '11.354256',
    'radius': '1000'
},
{
    'latitude' : '43.341757',
    'longitude': '11.305010',
    'radius': '1000'
},
{
    'latitude' : '43.343243',
    'longitude': '11.326733',
    'radius': '1000'
},
{
    'latitude' : '43.336999',
    'longitude': '11.300975',
    'radius': '1000'
},
{
    'latitude' : '43.323781',
    'longitude': '11.338891',
    'radius': '1000'
},
{
    'latitude' : '43.347800',
    'longitude': '11.326046',
    'radius': '1000'
},
{
    'latitude' : '43.332288',
    'longitude': '11.324176',
    'radius': '1000'
},
{
    'latitude' : '43.341445',
    'longitude': '11.312348',
    'radius': '1000'
},
{
    'latitude' : '43.326764',
    'longitude': '11.333702',
    'radius': '1000'
},
{
    'latitude' : '43.309896',
    'longitude': '11.315263',
    'radius': '1000'
},
{
    'latitude' : '43.317315',
    'longitude': '11.332998',
    'radius': '1000'
},
{
    'latitude' : '43.296856',
    'longitude': '11.334854',
    'radius': '1000'
},
{
    'latitude' : '43.343407',
    'longitude': '11.322849',
    'radius': '1000'
},
{
    'latitude' : '43.334640',
    'longitude': '11.304803',
    'radius': '1000'
}]

yelp_api_key = os.environ["YELP_API_KEY"]


yelp_url = "https://api.yelp.com/v3/businesses/search"


# headers = {
#   'Authorization': 'Bearer 7YecQFdZ8AtZibuTvODcWT6NQL2TCyh71cGZ-DbQjHkBbahj-qzC7p7WKcTw3EErXTNIfunMklRQmxzJnV9wAbwuL_Hv0yybKzXjcgMOMvBx89_Rqlncj6i4cKbEZHYx'
# }

yelp_api_key = 'Bearer %s' %yelp_api_key

headers = {
  'Authorization': yelp_api_key
}

params = {
    "latitude": 43.321600,
    "longitude": 11.327948,
    "radius": 1000
}

headers
payload = {}

# response = requests.request('GET', yelp_url, params=params, headers=headers, data=payload)
# res_json = response.json()

# res_json = res_json['businesses']
# res_json[0]['name']

yelp_res_json_lst = []
for params in yelp_params_lst:
    yelp_response = requests.get(yelp_url, params=params, headers=headers)
    yelp_res_json = yelp_response.json()
    yelp_res_json_lst.append(yelp_res_json)

yelp_res_json_lst[0]



# yelp_res_lst = []
# for yelp_res_json in yelp_res_json_lst:
#     yelp_result = yelp_res_json['businesses']
#     for i in range(len(yelp_result)):
#         dict = {}
#         dict['name'] = yelp_result[i]['name']
#         dict['address'] = yelp_result[i]['location']['address']
#         if yelp_result[i]['categories']:
#             dict['category'] = yelp_result[i]['categories'][0]['name']
#         yelp_res_lst.append(dict)

# yelp_res_lst

{'businesses': [{'id': 'TVssrfMd24fCSCsSq5JgFw',
   'alias': 'la-taverna-di-san-giuseppe-siena-2',
   'name': 'La Taverna di San Giuseppe',
   'image_url': 'https://s3-media2.fl.yelpcdn.com/bphoto/7-1SnE0SeFFitvoFDc2aeQ/o.jpg',
   'is_closed': False,
   'url': 'https://www.yelp.com/biz/la-taverna-di-san-giuseppe-siena-2?adjust_creative=0k5_VJSyIM2A1YDO2o8Iyw&utm_campaign=yelp_api_v3&utm_medium=api_v3_business_search&utm_source=0k5_VJSyIM2A1YDO2o8Iyw',
   'review_count': 159,
   'categories': [{'alias': 'tuscan', 'title': 'Tuscan'}],
   'rating': 5.0,
   'coordinates': {'latitude': 43.3154774, 'longitude': 11.3317009},
   'transactions': [],
   'price': '€€€',
   'location': {'address1': 'Via Giovanni Dupre 132',
    'address2': '',
    'address3': '',
    'city': 'Siena',
    'zip_code': '53100',
    'country': 'IT',
    'state': 'SI',
    'display_address': ['Via Giovanni Dupre 132', '53100 Siena', 'Italy']},
   'phone': '+39057742286',
   'display_phone': '+39 0577 42286',
   'distan

Parse through the response to get the POI (such as restaurants, bars, etc) details you want (ratings, name, location, etc)

In [18]:
yelp_res_lst = []
for yelp_res_json in yelp_res_json_lst:
    yelp_result = yelp_res_json['businesses']
    for i in range(len(yelp_result)):
        dict = {}
        dict['name'] = yelp_result[i]['name']
        dict['address'] = yelp_result[i]['location']['address1']
        dict['rating'] = yelp_result[i]['rating']
        if yelp_result[i]['categories']:
            dict['category'] = yelp_result[i]['categories'][0]['title']
        yelp_res_lst.append(dict)

yelp_res_lst

[{'name': 'La Taverna di San Giuseppe',
  'address': 'Via Giovanni Dupre 132',
  'rating': 5.0,
  'category': 'Tuscan'},
 {'name': 'Zest',
  'address': "Costa di Sant'antonio 13",
  'rating': 4.5,
  'category': 'Italian'},
 {'name': 'La Chiacchiera',
  'address': 'Via Costa di San Antonio 4',
  'rating': 4.0,
  'category': 'Italian'},
 {'name': 'Café Nannini',
  'address': 'Piazza Matteotti 32',
  'rating': 4.5,
  'category': 'Cafes'},
 {'name': 'Compagnia dei Vinattieri',
  'address': 'Via delle Terme, 79',
  'rating': 4.0,
  'category': 'Tuscan'},
 {'name': 'Salefino Vino e Cucina',
  'address': 'Via degli Umiliati 1',
  'rating': 4.5,
  'category': 'Tuscan'},
 {'name': 'Pizzeria alle Scalette',
  'address': 'Viale Curtatone 18',
  'rating': 4.5,
  'category': 'Pizza'},
 {'name': 'Il Bargello',
  'address': "Via di Citta' 53",
  'rating': 4.5,
  'category': 'Bars'},
 {'name': 'Antica Osteria da Divo',
  'address': 'Via Franciosa 25',
  'rating': 4.0,
  'category': 'Italian'},
 {'name

Put your parsed results into a DataFrame

In [20]:
df_yelp = pd.DataFrame(yelp_res_lst)
df_yelp

name                              address  rating  \
0    La Taverna di San Giuseppe               Via Giovanni Dupre 132     5.0   
1                          Zest             Costa di Sant'antonio 13     4.5   
2                La Chiacchiera           Via Costa di San Antonio 4     4.0   
3                  Café Nannini                  Piazza Matteotti 32     4.5   
4      Compagnia dei Vinattieri                  Via delle Terme, 79     4.0   
..                          ...                                  ...     ...   
218                  la veranda                      Piazza Calabria     4.5   
219                  Fontebecci                   Via Fiorentina 133     2.0   
220                 Mc Donald's                   Via Fiorentina 124     2.0   
221                   Bar Cecco  Viale Camillo Benso Conte Di Cavour     3.0   
222            Skile' Music Bar                   Piazza Calabria 18     3.0   

      category  
0       Tuscan  
1      Italian  
2      Italian  
3        Cafes  
4       Tuscan  
..         ...  
218    Italian  
219      Pizza  
220  Fast Food  
221      Cafes  
222      Cafes  

[223 rows x 4 columns]

# Comparing Results

Which API provided you with more complete data? Provide an explanation. 

In [26]:
df_foursquare.shape

(190, 3)

In [25]:
df_yelp.shape

(223, 4)

Yelp provides more data. It has 223 rows wihle Foursquare provides 199 rows

Get the top 10 restaurants according to their rating

In [28]:
df_yelp.sort_values(by='rating', ascending=False).head(10)

name                        address  rating  \
0      La Taverna di San Giuseppe         Via Giovanni Dupre 132     5.0   
138                  Ravioli Wang  Viale Vittorio Emanuele II 38     5.0   
168                L'oro di Siena               Via Garibaldi 61     5.0   
37                       Mugolone           Via dei Pellegrini 8     5.0   
40     La Taverna di San Giuseppe         Via Giovanni Dupre 132     5.0   
159  Antica Trattoria Botteganova         Strada Chiantigiana 29     5.0   
154                 Cartazucchero                Via Camollia 92     5.0   
47                 Duomo di Siena             Piazza del Duomo 8     5.0   
153                   La Mandorla                Via Camollia 36     5.0   
151                   Pasta Mondo               Via Camollia 173     5.0   

                             category  
0                              Tuscan  
138                           Chinese  
168                             Pizza  
37                            Italian  
40                             Tuscan  
159                           Italian  
154                    Used Bookstore  
47   Landmarks & Historical Buildings  
153                          Desserts  
151                             Pizza